tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py

```diff
@@ -94,17 +81,103 @@ def test_ignore_registry(key_string, key_object, tags_to_apply, get_configuratio
-                                               error_message='Did not receive expected '
-                                                             '"Ignoring ... due to( sregex)?" event').result()
-    ignored_registry = set(ignored_registry)
-    for ign_reg in sub_keys:
-        try:
-            ignored_registry.remove(os.path.join(keys_strings[0], ign_reg))
-        except KeyError:
-            raise KeyError(f'{os.path.join(keys_strings[0], ign_reg)} not in {ignored_registry}')
-    assert len(ignored_registry) == 0
+    Parameters
+    ----------
+    key_string : str
+        String name of the key.
+    key_object : object
+        Object winreg of the key.
+    tags_to_apply : set
+        Run test if matches with a configuration identifier, skip otherwise.
+    """
+    check_apply_test(tags_to_apply, get_configuration['tags'])
+    scheduled = get_configuration['metadata']['fim_mode'] == 'scheduled'
+    # Open the key (this shouldn't create an alert)
+    key_h = create_registry(registry_parser[root_key], registry, arch)
+    # Create values
+    modify_registry_value(key_h, value, REG_SZ, "test_value")
+    # Go ahead in time to let syscheck perform a new scan
+    check_time_travel(scheduled, monitor=wazuh_log_monitor)
+
+    event = wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                callback=callback_detect_event,
+                                error_message='Did not receive expected '
+                                                '"Sending FIM event: ..." event',
+                                accum_results=2 if triggers_event else 1).result()
+
+    if triggers_event:
+
+        assert event[0]['data']['type'] == 'modified', f'Parent key event type not equal'
+        assert event[0]['data']['path'] == os.path.join(root_key, registry), f'Wrong parent key path.'
+        assert event[0]['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', f'Parent key arch not equal.'
+
+        assert event[1]['data']['type'] == 'added', f'Wrong event type.'
+        assert event[1]['data']['path'] == os.path.join(root_key, registry), f'Wrong value path.'
+        assert event[1]['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', f'wrong key arch.'
+        assert event[1]['data']['value_name'] == value, f'Wrong value name'
+
+    else:
+        assert event['data']['type'] == 'modified', f'Parent key event type not equal'
+        assert event['data']['path'] == os.path.join(root_key, registry), f'Wrong parent key path.'
+        assert event['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', f'Parent key arch not equal.'
+
+        while True: # Look for the ignore event of the created value
+            ignored_value = wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                                   callback=callback_ignore).result()
```

Comment:
This line is under-indented.
```suggestion
                                                    callback=callback_ignore).result()
```